src/LEGO.AsyncAPI.Bindings/Sqs/Queue.cs

```diff
@@ -0,0 +1,116 @@
+
+        /// <summary>
+        /// The security policy for the SQS Queue
+        /// </summary>
+        public Policy Policy { get; set; }
+
+        /// <summary>
+        /// Key-value pairs that represent AWS tags on the topic.
+        /// </summary>
+        public Dictionary<string, string> Tags { get; set; }
+
+        public IDictionary<string, IAsyncApiExtension> Extensions { get; set; } = new Dictionary<string, IAsyncApiExtension>();
+
+        public static FixedFieldMap<Queue> fixedFieldMap => new()
+        {
+            { "name", (a, n) => { a.Name = n.GetScalarValue(); } },
+            { "fifoQueue", (a, n) => { a.FifoQueue = n.GetBooleanValue(); } },
+            { "deliveryDelay", (a, n) => { a.DeliveryDelay = n.GetIntegerValue(); } },
+            { "visibilityTimeout", (a, n) => { a.VisibilityTimeout = n.GetIntegerValue(); } },
+            { "receiveMessageWaitTime", (a, n) => { a.ReceiveMessageWaitTime = n.GetIntegerValue(); } },
+            { "messageRetentionPeriod", (a, n) => { a.MessageRetentionPeriod = n.GetIntegerValue(); } },
+            // temporarily removed for testing extensibility
+            // { "redrivePolicy", (a, n) => { a.RedrivePolicy = n.ParseMap(this.redrivePolicyFixedFields); } },
+            // { "policy", (a, n) => { a.Policy = n.ParseMap(this.policyFixedFields); } },
+            { "tags", (a, n) => { a.Tags = n.CreateSimpleMap(s => s.GetScalarValue()); } },
+        };
+        
+        public void Serialize(IAsyncApiWriter writer)
+        {
+            if (writer is null)
+            {
+                throw new ArgumentNullException(nameof(writer));
+            }
+
+            writer.WriteStartObject();
+            writer.WriteOptionalProperty(AsyncApiConstants.DollarRef, this.DollarRef);
+            writer.WriteRequiredProperty("name", this.Name);
+            writer.WriteOptionalProperty("fifoQueue", this.FifoQueue);
+            writer.WriteOptionalProperty("deliveryDelay", this.DeliveryDelay);
+            writer.WriteOptionalProperty("visibilityTimeout", this.VisibilityTimeout);
+            writer.WriteOptionalProperty("receiveMessageWaitTime", this.ReceiveMessageWaitTime);
+            writer.WriteOptionalProperty("messageRetentionPeriod", this.MessageRetentionPeriod);
+            writer.WriteOptionalObject("redrivePolicy", this.RedrivePolicy, (w, p) => p.Serialize(w));
+            writer.WriteOptionalObject("policy", this.Policy, (w, p) => p.Serialize(w));
+            writer.WriteOptionalMap("tags", this.Tags, (w, t) => w.WriteValue(t));
+            writer.WriteExtensions(this.Extensions);
+            writer.WriteEndObject();
+        }
+
+        public static Queue LoadQueue(ParseNode node)
+        {
+            var mapNode = node.CheckMapNode("queue");
+
+            var queue = new Queue();
+            foreach (var property in mapNode)
+            {
+                property.ParseField(queue, fixedFieldMap, ExtensionHelpers.GetExtensionsFieldMap<Queue>());
```

Comment:
@VisualBean Thanks for this! Been away but getting back into it now. Tried out your example but I'm getting an error here that `GetExtensionsFieldMap` is internal, it seems to work from within the test project but not in the separate bindings package. Does it need to be public?

Also, when you say

> 1 additional level of extensions

do you mean

- extensions within extensions are unsupported? or,
- extensions of an AsyncApiExtensible object within an AsyncApiExtensible object are unsupported?

Reply:
> extensions of an AsyncApiExtensible object within an AsyncApiExtensible object are unsupported?
this - atleast in the current 'simple' way of doing it.

Im creating an issue, to ensure all AsyncApiExtensible are automatically populated with any extensions, so we dont need to explicitly do it, in the readers.

Reply:
Ive made pre-releases for v4, so you can start using them. - will wait with the official release till these bindings are in.

Reply:
> Not being exhausted from renovating helps.

🥺 well I hope you have a beautiful home to live in now, thank you for still updating stuff while it's been happening.

I'll do some more today and see if I can match it up with the spec.